parsec/core/local_device.py

```diff
@@ -84,8 +97,12 @@ def generate_new_device(
 
 
 def get_key_file(config_dir: Path, device: LocalDevice) -> Path:
-    slug = device.slug
-    return get_devices_dir(config_dir) / slug / f"{slug}.keys"
+    """Return the keyfile path for a given device.
+
+    Note that the filename does not carry any intrinsic meaning.
+    Here, we simply use the slughash to avoid name collision.
+    """
+    return get_devices_dir(config_dir) / f"{device.slughash}.keys"
```

Comment:
Does this work with legacy key file when calling this function to retrieve the file in question ?
This is what is done in the GUI when changing the password...

Reply:
Oopsie, addressed here: 5017400cbeccb8427f9bdab2044dbeabeeade29c